### PR TITLE
[Ret] Last few fixes, all seals and Guardian, Alpha!

### PR DESCRIPTION
**Pull request by hillerstorm**
* Censure dot wasn't able to crit
* Hand of Light (mastery) didn't include Inquisition or 8% target dmg debuff
* Templar's Verdict was calculated in the wrong way
* Two-handed Weapon Spec didn't apply correctly to Judgements
* Add Seal of Insight, Seal of Justice and Seal of Righteousness
* Add travel time to Holy Wrath and Hammer of Wrath
* Add Guardian of Ancient Kings (!)
* Add auto reforge

* Mark as Alpha 😮

### CODE REVIEW

**Comment by hillerstorm** [comment]
Would really love for #670 to be merged so I can add the auto reforger. Without it I have to set the exp-cap to 4% because of our 10exp-glyph.

File: sim/paladin/guardian_of_ancient_kings.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	ancientFuryMinDamage, ancientFuryMaxDamage :=
+		core.CalcScalingSpellEffectVarianceMinMax(proto.Class_ClassPaladin, 0.2366, 0.3)
+	numTargets := paladin.Env.GetNumTargets()
+	results := make([]*core.SpellResult, numTargets)
+
+	ancientFury := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86704},
+		SpellSchool:    core.SpellSchoolHoly,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          core.SpellFlagAPL,
```
Comment (rosenrusinov) [inline sim/paladin/guardian_of_ancient_kings.go:147]: Is this spell castable manually? Prob dont want the APL flag in here

File: sim/paladin/guardian_of_ancient_kings.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+			},
+			CD: core.Cooldown{
+				Timer:    paladin.NewTimer(),
+				Duration: time.Minute * 5,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, unit *core.Unit, spell *core.Spell) {
+			goakAura.Activate(sim)
+			paladin.AncientGuardian.EnableWithTimeout(sim, paladin.AncientGuardian, duration)
```
Comment (rosenrusinov) [inline sim/paladin/guardian_of_ancient_kings.go:222]: You prob just need to enable the pet and dont need the timeout as you're manually handling the deactivation with the goak aura

File: sim/paladin/guardian_of_ancient_kings.go (hunk 1 above)
Comment (hillerstorm) [reply sim/paladin/guardian_of_ancient_kings.go:147]: No you're right, I shouldn't write code at 3 in the morning :)

File: sim/paladin/retribution/retribution.go [hunk 3]
```diff
@@ -72,6 +73,11 @@ func (ret *RetributionPaladin) RegisterSpecializationEffects() {
 	mhWeapon := ret.GetMHWeapon()
 	if mhWeapon != nil && mhWeapon.HandType == proto.HandType_HandTypeTwoHand {
 		ret.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexPhysical] *= 1.25
+		ret.AddStaticMod(core.SpellModConfig{
+			Kind:       core.SpellMod_DamageDone_Flat,
+			ClassMask:  paladin.SpellMaskJudgement,
+			FloatValue: 0.25,
+		})
```
Comment (rosenrusinov) [inline sim/paladin/retribution/retribution.go:80]: If there is a judgement that deals physical damage this and the above line would double dip this bonus. Not sure if that is possible but just worth mentioning

Comment (hillerstorm) [reply sim/paladin/retribution/retribution.go:80]: Hmm, they all deal holy damage but using melee hit/crit

File: sim/paladin/retribution/retribution.go [hunk 4]
```diff
@@ -83,22 +89,23 @@ func (ret *RetributionPaladin) RegisterMastery() {
...
-		ActionID:       actionId,
-		SpellSchool:    core.SpellSchoolHoly,
-		ProcMask:       core.ProcMaskMeleeMHSpecial,
-		Flags:          core.SpellFlagMeleeMetrics | core.SpellFlagIncludeTargetBonusDamage | core.SpellFlagNoOnCastComplete,
+		ActionID:    actionId,
+		SpellSchool: core.SpellSchoolHoly,
+		ProcMask:    core.ProcMaskMeleeMHSpecial,
+		Flags: core.SpellFlagMeleeMetrics |
+			// This flag makes sure 8% magic damage on target gets included as a multiplier
+			core.SpellFlagIncludeTargetBonusDamage |
```
Comment (rosenrusinov) [inline sim/paladin/retribution/retribution.go:97]: This comment is not true. That flag makes the spell include a very specific damage mod which is +bonus physical damage on target. You should remove this flag as its not corret on here.

From previous tests HoL should ignore both attacker and target modifiers and its only the 8% buff that affects it. Check Scourge Strike to see how to achieve that same effect

Comment (hillerstorm) [reply sim/paladin/retribution/retribution.go:97]: My bad, will fix

File: sim/paladin/ancient_guardian_pet.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		ancientGuardian.registerRetributionVariant()
+	} else if paladin.Spec == proto.Spec_SpecHolyPaladin {
+		ancientGuardian.registerHolyVariant()
+	}
+
+	ancientGuardian.PseudoStats.DamageTakenMultiplier = 0
+
+	// Guardians are not affected by raid buffs
+	ancientGuardian.PseudoStats.MeleeSpeedMultiplier = 1
+	ancientGuardian.PseudoStats.CastSpeedMultiplier = 1
```
Comment (rosenrusinov) [inline sim/paladin/ancient_guardian_pet.go:45]: I dont think you need this here as guardians already have checks to not have buffs applied to them. Only pets have a very little set of buffs applied to them

**Pull request merged**